### PR TITLE
feat: Handle new error code for guest links SQSERVICES-1049

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQSERVICES-1049" title="SQSERVICES-1049" target="_blank">StorySQSERVICES-1049</a>  [iOS] Use feature config to enable/disable guest links
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
There's a new error response
```
{
    code = 409;
    label = "guest-links-disabled";
    message = "The guest link feature is disabled and all guest links have been revoked.";
}

```
for `GET /conversations/conversationID/code` endpoint

that we should handle and do not display the alert message.

Needs releases with:

- [ wireapp/wire-ios-sync-engine/pull/1625 ]

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Conversation Options/ConversationOptionsViewController.swift [hunk 1]
```diff
@@ -96,7 +96,18 @@ final class ConversationOptionsViewController: UIViewController, UITableViewDele
...
+            present(UIAlertController.checkYourConnection(), animated: false)
+            return
+        }
+
+        switch error {
+        case .guestLinksDisabled:
+            break
+        case .noCode, .invalidOperation, .unknown:
+            present(UIAlertController.checkYourConnection(), animated: false)
+        }
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Conversation Options/ConversationOptionsViewController.swift:110]: One way you could simplify this:

```swift
if error == WirelessLinkError.guestLinksDisabled {
  return
} else {
  present(UIAlertController.checkYourConnection(), animated: false)
}
```

**Review by johnxnguyen — COMMENTED** [review]
There's a lint warning.

File: Wire-iOS/Sources/UserInterface/Conversation Options/ConversationOptionsViewController.swift (hunk 1 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Conversation Options/ConversationOptionsViewController.swift]: Done

**Comment by KaterinaWire** [comment]
> There's a lint warning.

Fixed

**Review by agisilaos — APPROVED** [review]
🚀

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**